pkg/parser/parser_audio.go

```diff
@@ -0,0 +1,103 @@
+package parser
+
+import (
+	"errors"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/readium/go-toolkit/pkg/asset"
+	"github.com/readium/go-toolkit/pkg/fetcher"
+	"github.com/readium/go-toolkit/pkg/internal/utils"
+	"github.com/readium/go-toolkit/pkg/manifest"
+	"github.com/readium/go-toolkit/pkg/mediatype"
+	"github.com/readium/go-toolkit/pkg/pub"
+)
+
+// Handles parsing of audiobooks from an unstructured archive format containing audio files, such as ZAB (Zipped Audio Book) or a simple ZIP.
+// It can also work for a standalone audio file.
+type AudioParser struct{}
+
+func (p AudioParser) Parse(asset asset.PublicationAsset, fetcher fetcher.Fetcher) (*pub.Builder, error) {
+	if !p.accepts(asset, fetcher) {
+		return nil, nil
+	}
+
+	links, err := fetcher.Links()
+	if err != nil {
+		return nil, err
+	}
+	readingOrder := make([]manifest.Link, 0)
+	for _, link := range links {
+		// Filter out all irrelevant files
+		fext := filepath.Ext(strings.ToLower(link.Href))
+		if len(fext) > 1 {
+			fext = fext[1:] // Remove "." from extension
+		}
+		_, contains := allowed_extensions_audio[fext]
+		if utils.IsHiddenOrThumbs(link.Href) || !contains {
+			continue
+		}
+		links = append(links, link)
+	}
+
+	if len(readingOrder) == 0 {
+		return nil, errors.New("no audio file found in the publication")
+	}
+
+	// Sort in alphabetical order
+	sort.Slice(readingOrder, func(i, j int) bool {
+		return readingOrder[i].Href < readingOrder[j].Href
+	})
```

Comment:
@mickael-menu could you provide me with a case where sorting would be wrong? I can't quite imagine a directory structure that would be an issue, since it depends whether we want folders to come before or after files, right?